pinot-timeseries/pinot-timeseries-spi/src/main/java/org/apache/pinot/tsdb/spi/series/BaseTimeSeriesBuilder.java

```diff
@@ -57,10 +58,24 @@ public BaseTimeSeriesBuilder(String id, @Nullable Long[] timeValues, @Nullable T
 
   public abstract void addValueAtIndex(int timeBucketIndex, Double value);
 
+  /**
+   * This is the method called by Pinot's leaf stage to accumulate data in the series builders. Pinot's leaf stage
+   * passes the raw time value to allow languages to build complex series builders. For instance, PromQL relies on
+   * the first and last time value in each time bucket for certain functions.
+   * <p>
+   *   The rawTimeValue is in the same Time Unit as that passed to the {@link LeafTimeSeriesPlanNode}.
+   * </p>
+   */
+  public abstract void addValueAtIndex(int timeBucketIndex, Double value, long rawTimeValue);
```

Comment:
self-review: by default we should fallback to the existing addValueAtIndex method and ignore the rawTimeValue.